homeassistant/components/sma_manager/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""SMA Manager Config Flow for UI configuration."""
+
+# Built-In Imports
+import logging
+import re
+
+# 3rd-Party Imports
+from sma_manager_api import SMA
+from voluptuous import Required, Schema
+
+# Home Assistant Imports
+from homeassistant import config_entries
+
+# Local Imports
+from .const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_REFRESH_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Enables the Integration to be configured in the UI."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Init Config Flow object with data default values."""
+        self._data = {
+            CONF_NAME: "SMA Manager",
+            CONF_HOST: "239.12.255.254",
+            CONF_PORT: 9522,
+            CONF_REFRESH_INTERVAL: 10,
+        }
+
+    def _get_schema(self) -> Schema:
+        """Generate a schema from data."""
+
+        return Schema(
+            {
+                Required(CONF_NAME, default=self._data[CONF_NAME]): str,
+                Required(CONF_HOST, default=self._data[CONF_HOST]): str,
+                Required(CONF_PORT, default=self._data[CONF_PORT]): int,
+                Required(
+                    CONF_REFRESH_INTERVAL, default=self._data[CONF_REFRESH_INTERVAL]
+                ): int,
```

Comment:
Why do we allow the user to configure this?

Reply:
Because he could be running multiple devices? Maybe he has different configs? Maybe he has a device that uses different stuff. I think it makes sense. The defaults are for my device and what most should run on but I wan't to allow for people to be able to change if needed.